pkg/virt-handler/vm.go

```diff
@@ -286,21 +286,24 @@ func handleDomainNotifyPipe(domainPipeStopChan chan struct{}, ln net.Listener, v
 	// Listen for new connections,
 	// Close listener and exit when stop encountered
 	go func(vmi *v1.VirtualMachineInstance, ln net.Listener, domainPipeStopChan chan struct{}) {
+		go func() {
+			<-domainPipeStopChan
+			log.Log.Object(vmi).Infof("closing notify pipe listener for vmi")
+			if err := ln.Close(); err != nil {
+				log.Log.Object(vmi).Infof("failed closing notify pipe listener for vmi: %v", err)
+			}
+		}()
 		for {
-			select {
-			case <-domainPipeStopChan:
-				log.Log.Object(vmi).Infof("closing notify pipe listener for vmi")
-				ln.Close()
-				return
-			default:
-				fd, err := ln.Accept()
-				if err != nil {
-					log.Log.Reason(err).Error("Domain pipe accept error encountered.")
-					// keep listening until stop invoked
-					time.Sleep(1)
-				} else {
-					fdChan <- fd
+			fd, err := ln.Accept()
```

Comment:
I understand the issue to be that we're trapped in this for loop and `<-domainPipeStopChan` never gets seen so the routine leaks.  Is that because we get stuck in the default clause?

Reply:
Correct, as `Accept()` is blocking, it would need to fail before the channel close gets observed.

Reply:
OK.  Probably more readable to have two different go routines instead of a nested one.  Also worth adding a comment explaining the importance of your change since it doesn't look like a unit test can cover what you fixed.

Reply:
Good point, I moved it out of here and added a comment to the error check below.